build-logic/src/main/java/qsl/internal/extension/QslModuleExtensionImpl.java

```diff
@@ -187,6 +190,29 @@ public void setupModuleDependencies() {
 		});
 	}
 
+	private void addTransitiveImplementations(QslLibraryDependency library, QslLibraryDependency.ModuleDependencyInfo info) {
```

Comment:
this is in desperate need of some code comments explaining the motivation behind this. i only can get my bearings because we talked about it in vc